courses-app/src/components/Registration/Registration.jsx

```diff
@@ -0,0 +1,78 @@
+import { Link, useHistory } from 'react-router-dom';
+import { useState } from 'react';
+
+import Button from '../common/Button';
+import Input from '../common/Input';
+
+import Service from '../../services/services';
+import './index.css';
+
+const Registration = () => {
+	const [inputsValue, setValues] = useState({});
+	const history = useHistory();
+	const service = new Service();
+
+	const handleInputName = ({ target }) => {
+		setValues({
+			...inputsValue,
+			[target.name]: target.value,
+		});
+	};
+
+	const onSubmit = async (event) => {
+		// const result = service.userLogin({ email, password });
+
+		// result.then((res) =>
+		// 	res.successful
+		// 		? history.push('/courses')
+		// 		: alert('Invalid password or email')
+		// );
+		event.preventDefault();
+		let { name = '', email = '', password = '' } = inputsValue;
+		const result = service.userRegistration({ name, email, password });
+
+		result.then((res) =>

```

Comment:
use async await instead of the promise chain

Reply:
thank you, good advice, done